### PR TITLE
Migration of MissingNewlineAtEndOfFile from AstNode.

### PR DESCRIPTION
**Pull request by lindamartin**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/MissingNewlineAtEndOfFileCheck.java [hunk 1]
```diff
@@ -41,18 +41,26 @@
   tags = {Tags.CONVENTION})
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("1min")
-public class MissingNewlineAtEndOfFileCheck extends SquidCheck<LexerlessGrammar> {
+public class MissingNewlineAtEndOfFileCheck extends BaseTreeVisitor {
+
+  private Logger LOG = LoggerFactory.getLogger(MissingNewlineAtEndOfFileCheck.class);
```
Comment (SonarTech) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/MissingNewlineAtEndOfFileCheck.java]: MAJOR Make the "LOG" logger private static final. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1312)
MAJOR Rename this field "LOG" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

File: javascript-checks/src/main/java/org/sonar/javascript/checks/MissingNewlineAtEndOfFileCheck.java [hunk 2]
```diff
@@ -41,18 +41,26 @@
...
+  private static final Logger LOG = LoggerFactory.getLogger(MissingNewlineAtEndOfFileCheck.class);
 
   @Override
-  public void visitFile(AstNode astNode) {
+  public void scanFile(AstTreeVisitorContext context) {
+    super.scanFile(context);
+
     RandomAccessFile randomAccessFile = null;
+
     try {
```
Comment (benzonico) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/MissingNewlineAtEndOfFileCheck.java]: We are now compatible with java 7 so you should probably use a try with resource.

**Comment by benzonico** [comment]
LGTM

File: javascript-checks/src/main/java/org/sonar/javascript/checks/MissingNewlineAtEndOfFileCheck.java (hunk 2 above)
Comment (lindamartin) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/MissingNewlineAtEndOfFileCheck.java]: @benzonico Indeed thanks, fixed in da0f5d84959e63d2bd76f402c26c932184738269.

**Comment by lindamartin** [comment]
Merged in e8bb6a1ebcd16ca1ba8af98bff3bc9c611c07269.

**Pull request closed without merging**